### ISSUE TITLE
Extract class probabilities

### ISSUE THREAD

**Issue by ruipimentelfigueiredo** [qubvel/segmentation_models.pytorch#726]
Hello
I would like to know, if possible, how to extract the probabilities of each class, for each pixel.
Thanks in advance

**Comment by da2r-20** [comment]
Can you elaborate a bit on what you are doing?
number of classes?
multiclass? multilabel?

**Comment by da2r-20** [comment]
```python
x_tensor  = torch.from_numpy(image).to(device).unsqueeze(0)
# I assume you are at inference time, no need for grads
with torch.no_grad():
    pr_mask   = model.predict(x_tensor)

pr_mask = (pr_mask.squeeze().cpu().numpy())
```
**Binary segmentation**
sigmoid on the logits will give you the probabilities.
If your last layer activation is None then you would have to apply it yourself

**Multiclass segmentation **
Same here but with softmax, You have to apply softmax on the output (logits)


And to answer your question, the output of the model is logits later transformed to output in [0, 1] or to a distribution by sigmoid, softmax activations respectively

good luck
